include/h2o/memory.h

```diff
@@ -199,7 +199,17 @@ void h2o_mem_clear_pool(h2o_mem_pool_t *pool);
 /**
  * allocates given size of memory from the memory pool, or dies if impossible
  */
-void *h2o_mem_alloc_pool(h2o_mem_pool_t *pool, size_t sz);
+void *h2o_mem_alloc_pool_aligned(h2o_mem_pool_t *pool, size_t alignment, size_t size);
+
+#ifdef H2O_NO_BUILT_IN_ALIGNOF
+#define H2O_ALIGNOF(type) (16)
+#else
+#define H2O_ALIGNOF(type) (__alignof__(type))
+#endif
+#define H2O_ALIGN(x, a) (((x) + (a)-1) & ~((a)-1))
```

Comment:
I would appreciate it if we could refrain from adding a public interface to calculate alignment. My understanding is that it is only used in memory.c.

Reply:
yes, will be fixed later.